### PR TITLE
chore(web): refactor step editor sidebar

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce?

Refactor the step editor sidebar to be a reusable component for all different steps, and reuse it in all the editors.
Refactor the workflow settings/snippet sidebar.
Implemented a separate `VariantsListSidebar` component for variants list.

### Other information (Screenshots)
Screenshot 2023-10-04 at 20 35 31

Screenshot 2023-10-04 at 20 36 00
Screenshot 2023-10-04 at 20 36 31
Screenshot 2023-10-04 at 20 36 42
Screenshot 2023-10-04 at 20 36 51

Screenshot 2023-10-04 at 21 15 30
Screenshot 2023-10-04 at 21 15 24
Screenshot 2023-10-04 at 21 15 19
Screenshot 2023-10-04 at 21 15 11
Screenshot 2023-10-04 at 21 15 04

### CODE REVIEW

File: apps/web/src/design-system/sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -78,6 +93,8 @@ export const Sidebar = ({
   isOpened,
   isExpanded = false,
   isLoading = false,
+  isParentScrollable = false,
```
Comment (LetItRock) [inline apps/web/src/design-system/sidebar/Sidebar.tsx:96]: allow to decide whether the whole sidebar is scrollable or only the body

File: apps/web/src/design-system/sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -106,41 +125,44 @@ export const Sidebar = ({
...
+        name="form-name"
+        noValidate
+        onSubmit={onSubmit}
+        data-test-id={dataTestId}
+        isParentScrollable={isParentScrollable}
+      >
         <HeaderHolder>
           {isExpanded && onBack && (
-            <ActionIcon variant="transparent" onClick={onBack} data-test-id="sidebar-back">
+            <ActionIcon onClick={onBack} data-test-id="sidebar-back">
```
Comment (LetItRock) [inline apps/web/src/design-system/sidebar/Sidebar.tsx:156]: the actions should have the background on hover

File: apps/web/src/pages/templates/components/ChannelStepEditor.tsx [hunk 3]
```diff
@@ -31,72 +26,29 @@ export const ChannelStepEditor = () => {
...
-        color={colors.white}
-        title={<StepName channel={channel} />}
-        style={{
-          width: '100%',
-          borderTopLeftRadius: 7,
-          borderBottomLeftRadius: 7,
-          paddingBottom: 24,
-        }}
-      >
+      <StepEditorSidebar>
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/ChannelStepEditor.tsx:29]: use the `StepEditorSidebar` for all editors

File: apps/web/src/pages/templates/components/ChannelStepEditor.tsx [hunk 4]
```diff
@@ -31,72 +26,29 @@ export const ChannelStepEditor = () => {
...
-      <SubPageWrapper
-        key={key}
-        color={colors.white}
-        title={<StepName channel={channel} />}
-        style={{ paddingBottom: 24 }}
-      >
-        {channel === StepTypeEnum.SMS && (
-          <>
-            <TemplateSMSEditor key={key} />
-            <TranslateProductLead id="translate-sms-editor" />
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/ChannelStepEditor.tsx:81]: moved the `TranslateProductLead` inside the editor

File: apps/web/src/pages/templates/components/StepNameInput.tsx [hunk 5]
```diff
@@ -26,8 +34,11 @@ export const StepNameInput = ({ path, defaultValue }: { path?: string; defaultVa
           <TextInput
             styles={(theme) => ({
               root: {
+                display: 'flex',
+                flexDirection: 'column-reverse',
+                gap: 4,
                 flex: '1 1 auto',
-                marginRight: 16,
+                width: '100%',
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/StepNameInput.tsx:41]: reverse the order of "step name" and the "step label" - for example:
```
Email
Variant 1
```

File: apps/web/src/pages/templates/components/StepNameLabel.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { Center } from '@mantine/core';
+import { useLocation } from 'react-router-dom';
+
+import { Variant, VariantsFile } from '../../../design-system/icons';
+import { Text, colors } from '../../../design-system';
+import { useStepIndex } from '../hooks/useStepIndex';
+
+export const StepNameLabel = ({ variantsCount = 0 }: { variantsCount?: number }) => {
```
Comment (LetItRock) [inline apps/web/src/pages/templates/components/StepNameLabel.tsx:8]: this is the smart label component that depending on the context renders the label:
- if root step - render nothing
- if variant - render `Variant {n}`
- if variants list - render `{n} variants`

**Review by ainouzgali — APPROVED** [review]
Beautiful

**Pull request merged**